### DISCUSSION TITLE
Visual-Inertial Odometry for a Minidrone (Project 234)

### DISCUSSION THREAD

**Discussion by robertogl** [mathworks/MATLAB-Simulink-Challenge-Project-Hub#68] [Collaborate]
Contribute to the discussion by asking and/or answering questions, commenting, or sharing your ideas for solutions to [project #234](https://github.com/mathworks/MathWorks-Excellence-in-Innovation/tree/main/projects/Visual%20-%20Inertial%20Odometry%20for%20a%20Minidrone)

---
*:robot: An AI bot may reply to your comments with suggestions and guidance. It will not solve the project for you. Your comments may be processed by a third-party LLM. To direct a question to a specific person, tag them with @username.*

**Comment by Rajat-Arora** [comment]
### Doubt 
Would it be beneficial to use a stereo camera instead of a monocular one to calculate visual-inertial odometry or is this already implemented?

**Reply by mordekarmaitreyee** [reply]
Hello @Rajat-Arora,

Thank you for the question. We are very happy to see you working on this project.

The hardware whose behavior is being modeled in simulations (Parrot Mambo) just has a monocular downward-facing camera. Considering the sensors that are present on the minidrone, a monocular camera would be necessary even though the idea of a stereo camera sounds great.

Please feel free to post questions here about the project. We will be happy to answer your questions.

Regards,
Maitreyee

**Reply by Rajat-Arora** [reply]
@mordekarmaitreyee  Thank you for your response, please guide me out of one more problem.

I entered my team size to be 2, so please let us know how shall/where shall
my teammate apply to officially be part of same team.

Thanks and regards
Rajat Arora

**Reply by mordekarmaitreyee** [reply]
Hello @Rajat-Arora,

Could you let me know the context of your question? The Excellence-in-Innovation projects are individual projects that can be taken up and would not be a team-based activity. 

Looking forward to seeing your work on the 'Visual - Inertial Odometry for a minidrone' project.

Thanks,
Maitreyee

**Reply by Anvesh-Som** [reply]
> Hello @Rajat-Arora,
> 
> Could you let me know the context of your question? The Excellence-in-Innovation projects are individual projects that can be taken up and would not be a team-based activity.
> 
> Looking forward to seeing your work on the 'Visual - Inertial Odometry for a minidrone' project.
> 
> Thanks, Maitreyee

Hi, @mordekarmaitreyee 

@Rajat-Arora and I were planning to tackle this project as a team, when we filled the [form](https://forms.office.com/Pages/ResponsePage.aspx?id=ETrdmUhDaESb3eUHKx3B5lOTzSa_A6lPqq2LJKzvpM5UMTBZRkc4UTRETjFERVRDWllQRE40OUFSQS4u) for Project 234 (Visual - Inertial Odometry for a minidrone), it had and entire page of "Team information" and asked number of team members, there it is mentioned that we give contacts of only POC, and the rest team members will "Receive the reward after submission" (if successful).

So our question was, how is the other member(s) suppose to register themselves and where? And if not, please let us know if it will be exclusively decided upon after submission of project.

Regards,
Anvesh Som

**Reply by robertogl** [reply]
Hi @Anvesh-Som and @Rajat-Arora,

Let me quickly chime in and clarify this. 
You can definitely tackle the project as a team! That is very common or almost the norm for school projects and we very much welcome teams working on our projects. We ask only one person to register and to serve as the point of contact between us and the team. Once you are ready to submit the solution, the point of contact will fill out a [submission form](https://forms.office.com/Pages/ResponsePage.aspx?id=ETrdmUhDaESb3eUHKx3B5lOTzSa_A6lPqq2LJKzvpM5UOFVCUTM4UTFYT0wwMDZPTjAzVFVMR1RXRy4u) where they will list the names and email addresses of all the team members so that we can send the rewards to all of them. Thus, to answer your question, the other members do not need to register as we will gather their information at the end of the project through the submission form. 

Please let me know if this answers your question and if you need anything else.

Roberto

**Reply by Anvesh-Som** [reply]
> Hi @Anvesh-Som and @Rajat-Arora,
> 
> Let me quickly chime in and clarify this. You can definitely tackle the project as a team! That is very common or almost the norm for school projects and we very much welcome teams working on our projects. We ask only one person to register and to serve as the point of contact between us and the team. Once you are ready to submit the solution, the point of contact will fill out a [submission form](https://forms.office.com/Pages/ResponsePage.aspx?id=ETrdmUhDaESb3eUHKx3B5lOTzSa_A6lPqq2LJKzvpM5UOFVCUTM4UTFYT0wwMDZPTjAzVFVMR1RXRy4u) where they will list the names and email addresses of all the team members so that we can send the rewards to all of them. Thus, to answer your question, the other members do not need to register as we will gather their information at the end of the project through the submission form.
> 
> Please let me know if this answers your question and if you need anything else.
> 
> Roberto

Hello @robertogl,
Thank you for clarifying all the points, I have very well understood it all.

Had one more question, is the hardware implementation on the exact drone specified by you (Parrot Mambo) a necessary part to consider the solution "complete"? Considering the fact that this drone has been discontinued by Parrot, and is barely available anywhere.

**Reply by robertogl** [reply]
Great question.

Hardware implementation is not a requirement for the success of the project. A working model in simulation would be enough. 
Of course, deployment on the actual hardware would validate the model with the new perception system but we are very aware of the issues you mentioned. 

The base Simulink model we provide is based on the parrot mini-drone. Are you planning to use a complete different model or focus only on the visual-odometry workflow?

Roberto

**Reply by mordekarmaitreyee** [reply]
Thank you @robertogl for helping with the question. And, apologies from my end for my statement. 

@Anvesh-Som and @Rajat-Arora, I hope your questions were answered. Please let us know if you have any questions while working on the project.

All the best!

-Maitreyee

**Comment by saiselvamngn** [comment]
Hi, I am Selvam Essaky, a Full-Time Ph.D. Student at Anna University, MIT Campus, Chennai, India. I am interested in working and collaborating on this project". Very happy and very grateful to the MathWorks research team for the opportunity.

**Comment by Yongoitsolo** [comment]
Hello, I was planning to pursue this project; however, I have a question - are hardware implementations required for validity and approval of the project?
I'd like to convey my thanks for providing us with this opportunity.

**Reply by robertogl** [reply]
Hi @Yongoitsolo,

Hardware validation is not required.

**Reply by Yongoitsolo** [reply]
Thank you for your reply. Understood.
